csrc/fusion.cpp

```diff
@@ -828,4 +837,34 @@ bool Fusion::hasDynamicTransform() {
   return !ir_utils::getTVsWithDynamicTransform(this).empty();
 }
 
+std::vector<Expr*> Fusion::getExprsToCodegen() {
+  std::vector<Expr*> exprs_requiring_codegen;
+  exprs_requiring_codegen.reserve(exprs().size());
+
+  for (auto expr : exprs()) {
+    bool is_expr_codegen = std::none_of(
+        expr->outputs().begin(), expr->outputs().end(), [this](Val* out) {
+          return this->getOutputAlias(out).type != AllocationType::Evaluate;
+        });
+
+    if (is_expr_codegen) {
+      exprs_requiring_codegen.emplace_back(expr);
+    }
+  }
```

Comment:
I don't follow the implementation. Shouldn't we return all exprs between inputs and codegen outputs like https://github.com/NVIDIA/Fuser/blob/925acc63e5870718ab83de911ec7f85d6126f729/csrc/device_lower/pass/expr_sort.cpp#L1524-L1525?

Reply:
I don't think this has been addressed in the latest version. IIUC, we need to return all exprs between inputs and codegen outputs instead of all exprs that define a codegen output. Wdyt?

Reply:
There's a difference between an Expr producing an EE output directly (as in your current code) and an Expr producing that indirectly (which is I think the right way). E.g. in

```
b = e0(a) // say codegen out
c = e1(b) // intermediate
d = e2(c) // say ee out
```

we do want to filter out `e1` even though its output is not **directly** marked for EE.

Reply:
Got it, thanks for the example.